src/server/bitops_family.cc

```diff
@@ -958,25 +958,24 @@ OpResult<std::vector<ResultType>> StateExecutor::Execute(const CommandList& comm
   return results;
 }
 
-nonstd::expected<CommonAttributes, std::string> ParseCommonAttr(CmdArgParser* prser) {
-  CmdArgParser& parser = *prser;
+nonstd::expected<CommonAttributes, std::string> ParseCommonAttr(CmdArgParser* parser) {
   CommonAttributes parsed;
   using nonstd::make_unexpected;
-  if (!parser.HasAtLeast(2)) {
+
+  auto [encoding, offset_str] = parser->Next<string_view, string_view>();
+
+  if (encoding.size() < 2) {
     return make_unexpected(kSyntaxErr);
   }
-
-  auto encoding = parser.ToUpper().Next();
-  if (absl::StartsWith(encoding, "U")) {
+  if (encoding[0] == 'U' || encoding[0] == 'u') {
```

Comment:
What's wrong with `absl::EqualsIgnoreCase` ? :eyes:

Reply:
string comparison is character by character. if you do `absl::EqualIgnoreCase(enconding[0], 'U')` it should be the exact same comparison under the hood

Reply:
EqualsIgnoreCase(absl::string_view piece1, absl::string_view piece2)
I don't see interface for char comparison

Reply:
a char is a `string_view` of size 1 :) Internally it should compare it character by character

Reply:
it's overkill)

Reply:
`std::string_view tmp = "t";` // tmp has now size 1

Reply:
> it's overkill)

I wouldn't say so but do whatever you like, I already accepted the PR :)